dc_signup_form/signup_server/wrappers.py

```diff
@@ -31,8 +31,12 @@ def get_users_payload(self, users):
 
     def add_users(self, payload):
         url = self.BASE_URL + "/contactdb/recipients"
-        r = requests.post(url, data=json.dumps(payload), headers=self.headers)
-        # if we get a non-2xx status code, just raise it
+        r = requests.post(url, data=json.dumps(payload), headers=self.headers) 
+        # SendGrid will return a 503 if "The API call was unsuccessful. 
+        # You should retry later." which is not a failure, 
+        # but we don't want to raise a sentry error for it
+        if r.status_code == 503:
+            return ""
```

Comment:
If this returns an empty string then the code that calls this method will fail.

`if response["error_count"] > 0`.

Basically, we're expecting a `dict` response from SendGrid, but this change isn't going to return that.

I think the best option here is to define a new exception class based on `ValueError` and catch that when we call the method:

```python

class SendGridAPIError(ValueError): pass
```

```python
try:
    response = sendgrid.add_users(
                sendgrid.get_users_payload([user.data for user in new_users])
            )
except SendGridAPIError:
    continue
```